pymc_experimental/model/marginal_model.py

```diff
@@ -244,6 +248,127 @@ def marginalize(self, rvs_to_marginalize: Union[TensorVariable, Sequence[TensorV
         # Raise errors and warnings immediately
         self.clone()._marginalize(user_warnings=True)
 
+    def unmarginalize(
+        self, idata, var_names=None, include_samples=False, extend_inferencedata=True
+    ):
+        """Computes log-likelihoods of marginalized variables conditioned on parameters
+        of the model given InferenceData with posterior group
+
+        Parameters
+        ----------
+        idata : InferenceData
+            InferenceData with posterior group
+        var_names : sequence of str, optional
+            List of Observed variable names for which to compute log_likelihood. Defaults to all observed variables
+        include_samples : bool, default False
+            Include samples of the marginalized variables
+        extend_inferencedata : bool, default True
+            Whether to extend the original InferenceData or return a new one
+
+        Returns
+        -------
+        idata : InferenceData
+            InferenceData with var_names added to posterior
+
+        """
+        if var_names is None:
+            var_names = self.marginalized_rvs
+
+        joint_logp = self.logp()
```

Comment:
`self.logp` will return the logp with the variables marginalized so they won't be part of the graph. I guess that's why you have `on_unused_inputs` issues later? I imagine you want the original logp that does not marginalize variables so you can give them values. 

That's why I was asking how you handle multiple related marginalized variables, seems like you have to either evaluate one at a time conditioned on the previous variables already evaluated or create a joint logp for all the combinations of the marginalized variables.

Reply:
Here is a test that creates nested marginalized variables: https://github.com/pymc-devs/pymc-experimental/blob/8046695e600970bb30a107376281d3e477a66dd0/pymc_experimental/tests/model/test_marginal_model.py#L169-L204

They get represented as an OpFromGraph where there are more than one output RVs without values. You can raise NotImplementedError for these cases for now so that you know you're working only with independent simple marginalized variables. Still I think you don't want to work with `self.logp`

Reply:
`self._logp` skips the marginalization step, maybe that's what you need?

Reply:
I chose to marginalise each discrete variable one at a time. Presumably I still need marginalise for that reason